code/game/objects/items/weapons/RSF.dm

```diff
@@ -232,6 +248,115 @@ RSF
 				to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
 				desc = "A RSF. It currently holds [matter]/30 fabrication-units."
 		return
+	else if(istype(A, /obj/structure/table) && mode == 7)
+		if(istype(A, /obj/structure/table) && matter >= 1)
+			to_chat(user, "Dispensing Snack - Tofu Turkey...")
+			playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+			new /obj/item/weapon/reagent_containers/food/snacks/tofurkey( A.loc )
+			if(isrobot(user))
+				var/mob/living/silicon/robot/engy = user
+				engy.cell.charge -= 500
+			else
+				matter--
+				to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
+				desc = "A RSF. It currently holds [matter]/30 fabrication-units."
+		return
+
+	else if(istype(A, /turf/simulated/floor) && mode == 7)
+		if(istype(A, /turf/simulated/floor) && matter >= 1)

```

Comment:
Redundant `istype`

Reply:
the structure of these is really mixed up.
Shouldn't you first check if `A` is the correct type of thing to dispense onto, then check which of the modes you are using?

As it stands, there's a LOT of duplicated code here